### ISSUE TITLE
Corrupted images in "file_result_storage" directory 

### ISSUE THREAD

**Issue by StepanKuksenko** [cshum/imagor#434]
Imagor stored a few corrupted images in the "file_result_storage" folder.
I've configured it to be cleaned after 24 hours, or if I remove it manually, a new correct image is served and stored.

We just started using Imagor in production, so I am very concerned about figuring out why this behavior occurred. 

Example:
![corrupted](https://github.com/cshum/imagor/assets/39428259/17c7e89c-60fe-41c6-ba99-a9a90dee1b5f)

After I removed from the cache:
![image](https://github.com/cshum/imagor/assets/39428259/d051895b-7973-4534-9599-fe9a02d1d32c)

**Comment by dlopez051** [comment]
Hello,

We have a similar problem:

Original image size: 32MB
Storage broken image size: 16MB
Desired size: 2000x400 with quality filter at 65%
Origen of image: web
Storage type (result and storage): file storage
Expiration: 2d
Imagor version: 1.4.12
And no logs available

Original
![2dd98ffb899f7baadf46eea5ddb1f648d57961eca44b95098a5b6d65_escalada](https://github.com/cshum/imagor/assets/145989712/6fc8e91b-f4d0-491e-842b-84160a3375f0)

Cache
![484fb722914f54ff48d21ff13319cf09a048_escalada](https://github.com/cshum/imagor/assets/145989712/b1b6290b-b2d2-4191-a46c-c9e0eb4aa094)

Result
![9e96bdb42025b8d7f5ccbded95fb7bd58262_escalada](https://github.com/cshum/imagor/assets/145989712/f317dab5-fa33-499a-8b89-24e7fffe00fb)

Maybe you can add a double check at the moment that you save the cache image that the image size is the same in the cache.

**Comment by thueske** [comment]
<img width="879" alt="Bildschirmfoto 2024-07-06 um 07 18 18" src="https://github.com/cshum/imagor/assets/436986/6b76bf7e-1d7b-40d7-baf5-18e1450fd188">

Same here. The log is empty (only the starting info).

**Comment by cshum** [comment]
Hi @StepanKuksenko  @dlopez051 @thueske  would you please test it under the docker build of master branch, and see if the corrupted image still show up as frequent?

https://github.com/cshum/imagor/pkgs/container/imagor/241055924?tag=master
```
ghcr.io/cshum/imagor@sha256:1d06522f4dcfd01ee9ef60c20f790015f9bfd374041e645d7d89827637194cd5
```

**Comment by StepanKuksenko** [comment]
thank y

> Hi @StepanKuksenko @dlopez051 @thueske would you please test it under the docker build of master branch, and see if the corrupted image still show up as frequent, or not anymore?
> 
> https://github.com/cshum/imagor/pkgs/container/imagor/241055924?tag=master
> 
> ```
> ghcr.io/cshum/imagor@sha256:1d06522f4dcfd01ee9ef60c20f790015f9bfd374041e645d7d89827637194cd5
> ```

thank you I will try

**Comment by thueske** [comment]
> Hi @StepanKuksenko @dlopez051 @thueske would you please test it under the docker build of master branch, and see if the corrupted image still show up as frequent, or not anymore?
> 
> https://github.com/cshum/imagor/pkgs/container/imagor/241055924?tag=master
> 
> ```
> ghcr.io/cshum/imagor@sha256:1d06522f4dcfd01ee9ef60c20f790015f9bfd374041e645d7d89827637194cd5
> ```

`no matching manifest for linux/arm64/v8 in the manifest list entries` 🙈

**Comment by cshum** [comment]
Oh only amd64 available for master branch docker, due to arm64 taking too long to build.
But I do want to verify the changes, before tagging a new version.

**Comment by dlopez051** [comment]
> Hi @StepanKuksenko @dlopez051 @thueske would you please test it under the docker build of master branch, and see if the corrupted image still show up as frequent, or not anymore?
> 
> https://github.com/cshum/imagor/pkgs/container/imagor/241055924?tag=master
> 
> ```
> ghcr.io/cshum/imagor@sha256:1d06522f4dcfd01ee9ef60c20f790015f9bfd374041e645d7d89827637194cd5
> ```

Hello,

Thanks for the change, I'm going to try it, in case I detect any similar images with this version, I'll let you know.

**Comment by cshum** [comment]
Fix available at v1.4.13
